drivers/gpio/gpio_nrfx.c

```diff
@@ -69,10 +69,6 @@ static void gpio_nrfx_gpd_retain_set(const struct device *port, uint32_t mask, g
 #ifdef CONFIG_SOC_NRF54H20_GPD
 	const struct gpio_nrfx_cfg *cfg = get_port_cfg(port);
 
-	if (cfg->pad_pd != NRF_GPD_FAST_ACTIVE1 || !(flags & GPIO_OUTPUT)) {
```

Comment:
Since the `flags` parameter is not to be used anymore, I think the function signature should be modified accordingly.